.github/workflows/industrial_ci_action.yml

```diff
@@ -0,0 +1,95 @@
+# This config uses industrial_ci (https://github.com/ros-industrial/industrial_ci.git).
+# For troubleshooting, see readme (https://github.com/ros-industrial/industrial_ci/blob/master/README.rst)
+
+name: BuildAndTest
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  industrial_ci:
+    strategy:
+      matrix:
+        env:
+          - IMAGE: 'foxy-ci'
+            CCOV: true
+            AFTER_RUN_TARGET_TEST: './.ci.prepare_codecov'
+          - IMAGE: 'foxy-ci-testing'
+            BEFORE_TARGET_TEST_EMBED: 'set +u && source moveit_kinematics/test/test_ikfast_plugins.sh && set -u'
+            CLANG_TIDY: true
+            CLANG_TIDY_BASE_REF: ${{ github.base_ref || github.ref }}
+    env:
+      DOCKER_IMAGE: moveit/moveit2:${{ matrix.env.IMAGE }}
+      UPSTREAM_WORKSPACE: moveit2.repos
+      CCOV: ${{ matrix.env.CCOV }}
+      AFTER_SETUP_UPSTREAM_WORKSPACE: 'vcs pull $BASEDIR/upstream_ws/src'
+      TARGET_CMAKE_ARGS: >
+        -DCMAKE_BUILD_TYPE=${{ matrix.env.CCOV && 'RelWithDebInfo' || 'Release'}}
+        ${{ matrix.env.CCOV && '-DCMAKE_CXX_FLAGS="--coverage" --no-warn-unused-cli' || '' }}
+      CCACHE_DIR: "${{ github.workspace }}/.ccache"
+      BASEDIR: ${{ github.workspace }}/.work
+      CACHE_PREFIX: "${{ matrix.env.IMAGE }}${{ matrix.env.CCOV && '-ccov' || '' }}"
+
+    name: "${{ matrix.env.IMAGE }}${{ matrix.env.CCOV && ' + ccov' || ''}}${{ matrix.env.CLANG_TIDY && ' + clang-tidy' || '' }}"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Free up disk space"
+        if: ${{ env.CCOV }}
+        run: |
+          sudo apt-get -qq purge build-essential ghc*
+          sudo apt-get clean
+          docker system prune -af
+          sudo mkdir /mnt/ccache
+          mkdir ${{ env.CCACHE_DIR }}
+          sudo mount --bind ${{ env.CCACHE_DIR }} /mnt/ccache
+          sudo rm -rf /usr/local/*
+          df -h
+      - uses: actions/checkout@v2
+      - name: cache upstream_ws
+        uses: pat-s/always-upload-cache@v2.1.3
+        with:
+          path: ${{ env.BASEDIR }}/upstream_ws
+          key: upstream_ws-${{ env.CACHE_PREFIX }}-${{ hashFiles('moveit2.repos') }}-${{ github.run_id }}
+          restore-keys: |
+            upstream_ws-${{ env.CACHE_PREFIX }}-${{ hashFiles('moveit2.repos') }}
+      - name: cache target_ws
+        if: ${{ ! env.CCOV }}
+        uses: pat-s/always-upload-cache@v2.1.3
+        with:
+          path: ${{ env.BASEDIR }}/target_ws
```

Comment:
You don't want to restore the src folder from cache. Otherwise you overwrite what you checked out, don't you?
Or is the checked out source symlinked into the source folder? If so, maybe add a comment here, to clarify this for our future selves.

Reply:
Can you comment on this, please?

Reply:
I left a comment on line 62-63